src/coreclr/src/vm/bundle.cpp

```diff
@@ -30,25 +30,25 @@ const SString &BundleFileLocation::Path() const
     return Bundle::AppBundle->Path();
 }
 
-Bundle::Bundle(LPCWSTR bundlePath, BundleProbe *probe)
+Bundle::Bundle(LPCSTR bundlePath, BundleProbe *probe)
 {
     STANDARD_VM_CONTRACT;
 
     _ASSERTE(probe != nullptr);
 
-    m_path.Set(bundlePath);
+    m_path.SetUTF8(bundlePath);
     m_probe = probe;
 
     // The bundle-base path is the directory containing the single-file bundle.
     // When the Probe() function searches within the bundle, it masks out the basePath from the assembly-path (if found).
 
-    LPCWSTR pos = wcsrchr(bundlePath, DIRECTORY_SEPARATOR_CHAR_W);
+    LPCSTR pos = strrchr(bundlePath, DIRECTORY_SEPARATOR_CHAR_A);
     _ASSERTE(pos != nullptr);
-    size_t baseLen = pos - bundlePath + 1; // Include DIRECTORY_SEPARATOR_CHAR_W in m_basePath
-    m_basePath.Set(bundlePath, (COUNT_T)baseLen);
+    size_t baseLen = pos - bundlePath + 1; // Include DIRECTORY_SEPARATOR_CHAR_A in m_basePath
+    m_basePath.SetUTF8(bundlePath, (COUNT_T)baseLen);
 }
 
-BundleFileLocation Bundle::Probe(LPCWSTR path, bool pathIsBundleRelative) const
+BundleFileLocation Bundle::Probe(SString path, bool pathIsBundleRelative) const
```

Comment:
We should also change the parameter type in Bundle::ProbeAppBundle below and make it `const SString&` - pretty much all the callers already pass in `SString` and so the current approach requires unnecessary conversions (and probably at least stack allocations).

Reply:
I have SString in first commit for BundleProbe, which I changed in second commit based on @janvorli's comment above: https://github.com/dotnet/runtime/pull/44466#discussion_r521263480.

Reply:
Hmm - I might be reading the code incorrectly. My understanding is that `SString` can store the string internally in UTF16 on all platforms. So with the current state of the change we go from `SString` in `AssemblyBinder::BindByTPAList` to `WCHAR*` which is probably a no-op as it returns the internal buffer. That is then converted back to `SString` which will make a copy (no conversion) and thus allocate.

But I'll defer to @janvorli 's opinion on this, he is much more experienced in this than me.

Reply:
Ah, I see. Now both `Probe` and `ProbeAppBundle` signatures are aligned.

Reply:
Why not `const SString&` - I'm not sure, but normal C++ semantics for simple `SString` would be to invoke copy ctor - which in SString case might actually allocate (It seems to have logic around mutable/immutable strings, but it's hard to tell if these will be mutable or not).

Reply:
Aligned `Probe` and `ProbeAppBundle` signatures on `const SString&`.

Reply:
SString can store three formats of string on all platforms. When the const WCHAR* operator is used, the SString is converted to UTF16 if it is not that way. Otherwise it just returns a pointer to the string. When SString is constructed over WCHAR*, it always allocates and copies the string in.
Assignment of const SString to SString seems to be considered immutable, so it should not allocate and it should use the original string until the new string is modified. So it seems like a better way.